python/cudf/cudf/tests/test_numba_import.py

```diff
@@ -19,10 +19,10 @@
 TEST_NUMBA_MVC_ENABLED = """
 import numba.cuda
 import cudf
-from cudf.utils._numba import _CUDFNumbaConfig, _patch_numba_mvc
+from cudf.utils._numba import _CUDFNumbaConfig, patch_numba_linker_cuda_11
 
 
-_patch_numba_mvc()
+patch_numba_linker_cuda_11()
```

Comment:
This seems to be unconditionally using CUDA 11 patching logic in tests. We should use CUDA 12 patching logic when on CUDA 12, right?

Reply:
I think this whole test skips in a CUDA 12 environment based on the below
```
@pytest.mark.skipif(
    not IS_CUDA_11, reason="Minor Version Compatibility test for CUDA 11"
)
```

Reply:
The MVC tests should be enabled, even if we don't have an MVC-triggering environment in CI with the drivers/runtimes we use. We want to be able to run `pytest` on a system requiring MVC and know that it's being covered by cudf.

Reply:
It's okay to have them be skipped if on CUDA 12 `and` MVC is needed `and` pynvjitlink isn't available.

Reply:
Ok, that makes sense. I'm thinking I'll rename this file `test_mvc` and have a separate test for cuda 11 and cuda 12. updates to follow.

Reply:
did some refactoring in https://github.com/rapidsai/cudf/pull/13650/commits/35b7e59e002c7c1bf07eb61063cb86a7c61c6577